src/main/java/io/jenkins/plugins/okhttp/api/internals/JenkinsProxySelector.java

```diff
@@ -20,23 +20,23 @@
 @Restricted(NoExternalUse.class)
 public class JenkinsProxySelector extends ProxySelector {
     private static final Logger LOGGER = Logger.getLogger(JenkinsProxySelector.class.getName());
-    private final ProxyConfiguration configuration;
-    private final Proxy proxy;
-
-    public JenkinsProxySelector(final ProxyConfiguration configuration) {
-        this.configuration = configuration;
-        this.proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(configuration.name, configuration.port));
-    }
 
     @Override
     public List<Proxy> select(URI uri) {
+
+        ProxyConfiguration configuration = Jenkins.get().proxy;
```

Comment:
use the getter (available in 2.205+)

```suggestion
        ProxyConfiguration configuration = Jenkins.get().getProxy();
```